### PR TITLE
Remove analyzer suppressions

### LINKED ISSUE #17154: [ServiceBus] Re-enable suppressed warnings
Suppressions for CA1816;CA2016 were added as part of FxCop rules update. Individual callsites need to be investigated and fixed/suppressed.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/17154

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusLiveTestBase.cs [hunk 1]
```diff
@@ -40,7 +40,6 @@ protected ServiceBusClient CreateClient(int tryTimeout = DefaultTryTimeout)
                     RetryOptions = new ServiceBusRetryOptions
                     {
                         TryTimeout = TimeSpan.FromSeconds(tryTimeout),
-                        MaxRetries = 0
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusLiveTestBase.cs:43]: This was a copy paste error added in https://github.com/Azure/azure-sdk-for-net/pull/20940

**Review by jsquire — APPROVED** [review]

**Pull request merged**